src/Lucene.Net/Search/FieldDocSortedHitQueue.cs

```diff
@@ -102,27 +102,54 @@ public override bool LessThan(FieldDoc docA, FieldDoc docB)
 				int type = fields.Array[i + fields.Offset].Type;
 				if(type == SortField.STRING)
 				{
-				    string s1 = (string) docA.fields[i];
-                    string s2 = (string) docB.fields[i];
-                    // null values need to be sorted first, because of how FieldCache.getStringIndex()
-                    // works - in that routine, any documents without a value in the given field are
-                    // put first.  If both are null, the next SortField is used
-                    if (s1 == null)
+                    if (docA.fields[i] is UnmanagedStringArray.UnmanagedString us1 &&
+                        docB.fields[i] is UnmanagedStringArray.UnmanagedString us2)
                     {
-                        c = (s2 == null) ? 0 : -1;
-                    }
-                    else if (s2 == null)
-                    {
-                        c = 1;
-                    }
-                    else if (fields.Array[i + fields.Offset].Locale == null)
-                    {
-                        c = s1.CompareTo(s2);
+                        // null values need to be sorted first, because of how FieldCache.getStringIndex()
+                        // works - in that routine, any documents without a value in the given field are
+                        // put first.  If both are null, the next SortField is used
+                        if (us1.IsNull)
+                        {
+                            c = (us2.IsNull) ? 0 : -1;
+                        }
+                        else if (us2.IsNull)
+                        {
+                            c = 1;
+                        }
+                        else if (fields.Array[i + fields.Offset].Locale == null)
+                        {
+                            c = us1.CompareTo(us2);
+                        }
+                        else
+                        {
+                            c = collators[i].Compare(us1.ToString(), us2.ToString());
```

Comment:
Under what scenario do we have this?

Reply:
Not sure TBH... this seems to be culture depended, so I left it as is..

Reply:
Pretty sure that this isn't used at all in our usecase

Reply:
3 tests failed when I throw in this code path.
`TestInternationalMultiSearcherSort`
`TestMultiSort`
`TestParallelMultiSort`

Reply:
But I don't see any usages in ravendb itself for the `Locale`

Reply:
Yes, we don't care about this for Raven